libsignal-service/src/sender.rs

```diff
@@ -750,7 +751,25 @@ where
         recipient: Option<&ServiceAddress>,
         data_message: Option<crate::proto::DataMessage>,
         timestamp: u64,
+        send_message_results: &[&SendMessageResult],
```

Comment:
If you make this `&[SendMessageResult]`, you can just borrow the `Vec` above instead of creating a new vec with only references. The extra `&` inside the slide doesn't really give you anything extra useful in terms of reference.